code/modules/fishing/fishing_net.dm

```diff
@@ -0,0 +1,106 @@
+/obj/item/weapon/material/fishing_net
+	name = "fishing net"
+	desc = "A crude fishing net."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "net"
+	item_state = "net"
+
+	var/empty_state = "net"
+	var/contain_state = "net_full"
+
+	w_class = ITEMSIZE_SMALL
+	flags = NOBLUDGEON
+
+	slowdown = 0.25
+
+	reach = 2
+
+	default_material = "cloth"
+
+	var/list/accepted_mobs = list(/mob/living/simple_mob/animal/passive/fish)
+
+/obj/item/weapon/material/fishing_net/Initialize()
+	..()
+	update_icon()
+
+/obj/item/weapon/material/fishing_net/afterattack(var/atom/A, var/mob/user, var/proximity)
+	if(get_dist(get_turf(src), A) > reach)
+		return
+
+	if(istype(A, /turf))
+		var/mob/living/Target
+		for(var/type in accepted_mobs)
+			Target = locate(type) in A.contents
+			if(Target)
+				afterattack(Target, user, proximity)
+				break
+
+	if(istype(A, /mob))
+		var/accept = FALSE
+		for(var/D in accepted_mobs)
```

Comment:
`if(A.type in accepted_mobs)`

Reply:
That would exempt subtypes, which is why this uses an istype on a for loop of [in the only existing subtype] one item.